plutus-core/plutus-core/src/PlutusCore/ParserCommon.hs

```diff
@@ -69,64 +54,65 @@ intern n = do
             put $ ParserState identifiers'
             return fresh
 
-type Parser = ParsecT ParseError T.Text (StateT ParserState PLC.Quote)
-instance (Stream s, PLC.MonadQuote m) => PLC.MonadQuote (ParsecT e s m)
-
-parse :: Parser a -> String -> T.Text -> Either (ParseErrorBundle T.Text ParseError) a
+parse :: Parser SourcePos a -> String -> T.Text -> Either (ParseErrorBundle T.Text (PLC.ParseError SourcePos)) a
 parse p file str = PLC.runQuote $ parseQuoted p file str
 
-parseQuoted :: Parser a -> String -> T.Text -> PLC.Quote (Either (ParseErrorBundle T.Text ParseError) a)
+parseQuoted :: Parser SourcePos a -> String -> T.Text -> PLC.Quote
+                   (Either (ParseErrorBundle T.Text (PLC.ParseError SourcePos)) a)
 parseQuoted p file str = flip evalStateT initial $ runParserT p file str
 
-whitespace :: Parser ()
+whitespace :: Parser SourcePos ()
 whitespace = Lex.space space1 (Lex.skipLineComment "--") (Lex.skipBlockCommentNested "{-" "-}")
 
-lexeme :: Parser a -> Parser a
+lexeme :: Parser SourcePos a -> Parser SourcePos a
 lexeme = Lex.lexeme whitespace
 
-symbol :: T.Text -> Parser T.Text
+symbol :: T.Text -> Parser SourcePos T.Text
 symbol = Lex.symbol whitespace
 
-lparen :: Parser T.Text
+lparen :: Parser SourcePos T.Text
 lparen = symbol "("
-rparen :: Parser T.Text
+rparen :: Parser SourcePos T.Text
 rparen = symbol ")"
 
-lbracket :: Parser T.Text
+lbracket :: Parser SourcePos T.Text
 lbracket = symbol "["
-rbracket :: Parser T.Text
+rbracket :: Parser SourcePos T.Text
 rbracket = symbol "]"
 
-lbrace :: Parser T.Text
+lbrace :: Parser SourcePos T.Text
 lbrace = symbol "{"
-rbrace :: Parser T.Text
+rbrace :: Parser SourcePos T.Text
 rbrace = symbol "}"
 
-inParens :: Parser a -> Parser a
+inParens :: Parser SourcePos a -> Parser SourcePos a
 inParens = between lparen rparen
 
-inBrackets :: Parser a -> Parser a
+inBrackets :: Parser SourcePos a -> Parser SourcePos a
 inBrackets = between lbracket rbracket
 
-inBraces :: Parser a -> Parser a
+inBraces :: Parser SourcePos a-> Parser SourcePos a
 inBraces = between lbrace rbrace
 
 isIdentifierChar :: Char -> Bool
 isIdentifierChar c = isAlphaNum c || c == '_' || c == '\''
 
+stringLiteral :: Parser SourcePos String
+stringLiteral = char '"' >> manyTill Lex.charLiteral (char '"')
```

Comment:
Not sure if we care about performance of the parser, but this is super inefficient. `takeWhileP` [can be orders of magnitude faster](https://markkarpov.com/post/megaparsec-more-speed-more-power.html#there-is-hope) than `manyTill` on `Text`.

Reply:
Sorry I think the function is in there by mistake. We don't actually use that now so I deleted it altogether.